src/main/scala/firrtl/passes/Passes.scala

```diff
@@ -125,13 +125,13 @@ object ExpandConnects extends Pass {
           case sx: DefMemory => genders(sx.name) = MALE; sx
           case sx: DefNode => genders(sx.name) = MALE; sx
           case sx: IsInvalid =>
-            val invalids = (create_exps(sx.expr) foldLeft Seq[Statement]())(
-               (invalids,  expx) => gender(set_gender(expx)) match {
-                  case BIGENDER => invalids :+ IsInvalid(sx.info, expx)
-                  case FEMALE => invalids :+ IsInvalid(sx.info, expx)
-                  case _ => invalids
+            val invalids = create_exps(sx.expr).flatMap { case expx =>
```

Comment:
Is flatMap fast?

Reply:
The answer is yes but now I'm doing some microbenchmarking and foldLeft with List prepend followed by a reverse might actually be faster...

Reply:
I think more digging is necessary, but foldLeft with list prepend then a reverse might be a bit faster for simple iteration, but I think flatMap is faster when you're combining Seqs (eg. in create_exps)